src/card/view/CardView.tsx

```diff
@@ -130,7 +131,18 @@ const NeoCardView = ({
     if (!settingsOpen) {
       setLastRunTimestamp(Date.now());
     }
-  }, [settingsOpen, query, JSON.stringify(localParameters)]);
+  }, [JSON.stringify(localParameters)]);
+
+  useEffect(() => {
+    if (!settingsOpen && queryChange) {
+      setLastRunTimestamp(Date.now());
+    }
+    setQueryChange(false);
+  }, [settingsOpen]);
+
+  useEffect(() => {
+    setQueryChange(true);
```

Comment:
Looks like this sometimes breaks queries that re-run on refresh rate

Reply:
Can you elaborate on the sometimes? Im trying and its not failing :/